autogluon/utils/tabular/ml/learner/default_learner.py

```diff
@@ -105,7 +117,120 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, scheduler_options=None, hy
         time_end = time.time()
         self.time_fit_training = time_end - time_preprocessing_end
         self.time_fit_total = time_end - time_preprocessing_start
-        logger.log(20, f'AutoGluon training complete, total runtime = {round(self.time_fit_total, 2)}s ...')
+        if X_test_unlabeled is not None:
+            logger.log(20, f'Initial training (without test data) complete after {round(self.time_fit_total, 2)}s')
+            if time_limit:
+                remaining_time = self.time_limit - self.time_fit_total
+            else:
+                remaining_time = None
+            self.transductive_fit(X=X_og, X_val=X_val_og, X_test_unlabeled=X_test_unlabeled, scheduler_options=scheduler_options, hyperparameter_tune=hyperparameter_tune,
+                                  feature_prune=feature_prune, holdout_frac=holdout_frac, num_bagging_folds=num_bagging_folds, num_bagging_sets=num_bagging_sets, stack_ensemble_levels=stack_ensemble_levels,
+                                  hyperparameters=hyperparameters, ag_args_fit=ag_args_fit, excluded_model_types=excluded_model_types, time_limit=remaining_time,
+                                  save_data=save_data, save_bagged_folds=save_bagged_folds, verbosity=verbosity, feature_generator=joint_feature_generator)
+        else:
+            logger.log(20, f'AutoGluon training complete, total runtime = {round(self.time_fit_total, 2)}s')
+
+    def transductive_fit(self, X: DataFrame, X_val: DataFrame, X_test_unlabeled: DataFrame,
+                         scheduler_options, hyperparameter_tune, feature_prune, holdout_frac, num_bagging_folds, num_bagging_sets, stack_ensemble_levels,
+                         hyperparameters, ag_args_fit, excluded_model_types, time_limit, save_data, save_bagged_folds, verbosity,
+                         feature_generator, pseudolabel=True, detect_shift=False):
+        """ After initial fit, transductive_fit will continue training additional models using additional unlabeled test data (X_test_unlabeled).
+            Arguments are the same as Learner.fit().
+            Additional arguments:
+                pseudolabel (bool): whether to utilize pseudolabeling
+                detect_shift (bool): whether to correct for distribution-shift between training/test data distributions.
+
+            TODO: add hyperparameter to only retain confident pseudolabels.
```

Comment:
without using softclass or thresholding, I don't think this is usable, as it will frequently harm results.